### PR TITLE
Add Mattermost Desktop 4.7.1

### PR DESCRIPTION
**Pull request by al-lac**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/23707)

### CODE REVIEW

File: manifests/m/Mattermost/MattermostDesktop/4.7.1/Mattermost.MattermostDesktop.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- silent
+- silentWithProgress
+Installers:
+- InstallerLocale: en-US
+  Architecture: x64
+  InstallerType: msi
+  Scope: machine
+  InstallerUrl: https://releases.mattermost.com/desktop/4.7.1/mattermost-desktop-4.7.1-x64.msi
+  InstallerSha256: b606a5c5067c41f113eb2b2955cd4ae6ade2e0952849fc3105012e295a12ba97
+  UpgradeBehavior: install
```
Comment (jedieaston) [inline manifests/m/Mattermost/MattermostDesktop/4.7.1/Mattermost.MattermostDesktop.installer.yaml:20]:
```suggestion
  UpgradeBehavior: install
  ProductCode: "{D093999D-3BE8-4206-BBD9-73F2AB83847D}"
```

File: manifests/m/Mattermost/MattermostDesktop/4.7.1/Mattermost.MattermostDesktop.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: msi
+  Scope: machine
+  InstallerUrl: https://releases.mattermost.com/desktop/4.7.1/mattermost-desktop-4.7.1-x86.msi
+  InstallerSha256: dc87787a1a602b6a2a3eaaaf4d3bf073548caa350855030f88ce3ccb3dda76db
+  UpgradeBehavior: install
```
Comment (jedieaston) [inline manifests/m/Mattermost/MattermostDesktop/4.7.1/Mattermost.MattermostDesktop.installer.yaml:27]:
```suggestion
  UpgradeBehavior: install
  ProductCode: "{2A61AC04-EB7F-44D6-B33B-45E5D8A56838}"
```

**Review by jedieaston — CHANGES_REQUESTED** [review]
Added the product codes to assist with matching. 

Hot tip: The latest version of wingetcreate has support to do this automatically. When you have a minute, do `winget upgrade wingetcreate`.

**Comment by al-lac** [comment]
Hi @jedieaston, thanks! Will use the updated version of wingetcreate next time!

**Review by OfficialEsco — APPROVED** [review]

**Pull request merged**